source/exercises/catalog.md

```diff
@@ -94,12 +103,307 @@ USGS 网站提供了在线搜索和查看地震目录的功能，既可以手动
+```
+
+在需要时，随时可以使用 {func}`read_events() <obspy.core.event.read_events>` 函数读入
+磁盘文件中的地震目录。该函数值返回 {class}`~obspy.core.event.Catalog` 类型：
+```{code-cell} ipython3
+from obspy import read_events
+
+cat = read_events("japan-earthquakes.xml")
+print(cat)
+```
+
+## 深入理解和使用 {class}`~obspy.core.event.Catalog` 类
+
+上面提到，{meth}`Client.get_events() <obspy.clients.fdsn.client.Client.get_events()>`
+和 {func}`read_events() <obspy.core.event.read_events>` 的返回值都是
+{class}`~obspy.core.event.Catalog` 类型。
+
+事实上，{class}`~obspy.core.event.Catalog` 类是 ObsPy 中最核心的类之一，其用于储存
+地震目录信息。下图展示了 {class}`~obspy.core.event.Catalog` 类的属性及其层级关系：
+
+:::{figure} https://docs.obspy.org/_images/Event.png
+:align: center
+:alt: "ObsPy 的 Catalog 类"
+:width: 100%
+
+ObsPy 的{class}`~obspy.core.event.Catalog` 类。引自 [ObsPy 网站](https://docs.obspy.org/_images/Event.png)。
+:::
+
+{class}`~obspy.core.event.Catalog` 类可以当作一个列表。
+像常规列表一样，我们可以对 {class}`~obspy.core.event.Catalog` 类里的地震事件进行循环：
+````{margin}
+```{note}
+`for event in cat:` 会输出所有地震的信息。为了节省空间，这里只输出了前两个地震的信息。
+```
+````
+```{code-cell} ipython3
+for event in cat[0:2]:
+    print(event)
+```
+
+{class}`~obspy.core.event.Catalog` 列表里的每个元素都是 {class}`~obspy.core.event.Event` 类型。
+下面以第一个事件为例，看看 {class}`~obspy.core.event.Event` 类里的内容：
+```{code-cell} ipython3
+event = cat[0]
+print(event)
+```
+
+从中可以看出，{class}`~obspy.core.event.Event` 类有很多属性。在这一节里，
+我们重点关注 `origins` 和 `magnitudes`。
+
+{class}`~obspy.core.event.Event` 的 `origins` 属性也是一个列表，其元素是 {class}`~obspy.core.event.origin.Origin` 类型。
+```{code-cell} ipython3
+print(event.origins)
+```
+
+`event.origins` 之所以是一个列表，是因为，对于任意一个地震，可能有多个机构或多种不同方法给出多个不同的震源信息。
+在这个例子中，`event.origins` 中只包含了一个元素（即一个震源信息）。
+```{code-cell} ipython3
+origin = event.origins[0]
+print(origin)
+```
+
+从上面的输出中可以看到，{class}`~obspy.core.event.origin.Origin` 类的属性中包含了我们关心的震源
+信息。比如，可以通过下面的代码，输出地震震源的发震时刻、纬度、经度和深度信息：
+````{margin}
+```{note}
+ObsPy 中震源深度的单位为 m，而有些地震目录中深度的单位为 km。在使用时需要格外注意。
+```
+````
+```{code-cell} ipython3
+print(origin.time, origin.latitude, origin.longitude, origin.depth)
+```
+
+同样的，{class}`~obspy.core.event.Event` 的 `magnitudes` 属性也是一个列表，
+其元素是 {class}`~obspy.core.event.magnitude.Magnitude` 类型。
+```{code-cell} ipython3
+print(event.magnitudes)
+```
+```{code-cell} ipython3
+mag = event.magnitudes[0]
+print(mag)
+```
+下面的代码将输出地震震级和震级类型信息：
+```{code-cell} ipython3
+print(mag.mag, mag.magnitude_type)
+```
+
 ## 使用地震目录绘图
 
-### 地震分布图
+在获取地震目录之后，通常还需要对地震目录做一些简单的绘图和分析。这一小节演示如何
+绘制最简单的地震深度直方图、地震震级直方图以及震级-频次关系图。
+
+首先，需要把地震目录准备好。这里我们选择下载 2000 - 2010 年间震级大于 4.8 级，
+震源深度大于 50 km 的地震。这一地震目录中共计约 7000 个地震：
+```{code-cell} ipython3
+client = Client("USGS")
+cat = client.get_events(
+    starttime="2000-01-01",
+    endtime="2010-01-01",
+    minmagnitude=4.8,
+    mindepth=50,
+)
+print(cat)
+```
+
+为了进一步对数据做处理以及绘图，我们还需要导入 [NumPy](https://numpy.org/)
+和 [Matplotlib](https://matplotlib.org/) 模块：
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+import numpy as np
+```
 
 ### 地震深度分布直方图
```

Comment:
I feel the following two subsections could be moved to `data analysis` sections. We should focus on catalog downloading in this section, otherwise, it's too long. The readers might lose themselves.

Reply:
Done.